diagnostics_tools/include/diagnostic_tools/periodic_event_status.h

```diff
@@ -76,18 +76,21 @@ class PeriodicEventStatusParams final {
 
 class PeriodicEventStatus : public PeriodicDiagnosticTask {
  public:
-  PeriodicEventStatus(const std::string &name);
+  PeriodicEventStatus(const std::string &name, const size_t &last_cycle_period_avg_window,
+                      const size_t &historic_period_avg_window);
 
-  PeriodicEventStatus(const std::string &name, PeriodicEventStatusParams params);
+  PeriodicEventStatus(const std::string &name, PeriodicEventStatusParams params,
+                      const size_t &last_cycle_period_avg_window,
+                      const size_t &historic_period_avg_window);
```

Comment:
@patapufate consider putting window sizes into the params structure and providing reasonable defaults. Also, being PODs, I'd suggest you remove `const` and `&` qualifiers.